metadata.yaml

```diff
@@ -45,14 +45,19 @@ provides:
     optional: true
   database:
     interface: postgresql_client
+    optional: true
   db:
     interface: pgsql
+    optional: true
   db-admin:
     interface: pgsql
+    optional: true
```

Comment:
I am not sure here. From one side, the charm can be deployed and will be active and fully functional without any relations... while technically user have to relate one of database/db/db-admin endpoint. It is not possible to use charm otherwise, from this point of view: they are mandatory (but mutually exclusive).

Why do you need optional here?

P.S. I am fine to mark optional metrics-endpoint/grafana-dashboard.

Reply:
I've removed them here. The motivation for the change is for automated charm integration testing where these flags are used to discover the minimally deployable bundle for the charm to run. Since the metadata yaml doesn't support mutual exclusion (yet? eventually to be solved with scriplets), SQA has a way to hard code that.